### ISSUE TITLE
Cannot do anything without `native-client` feature

### ISSUE THREAD

**Issue by pbzweihander** [rustasync/surf#41]
Since clients are all `pub(crate)` and `one_off` module is depend on `native-client` feature, we can't do anything without `native-client` feature. We should make other clients public or used by `one_off` module.

**Comment by CryZe** [comment]
Yeah I just ran into this too. It basically makes it so you can't use the same code on both wasm and native, which I thought was the whole point of this library.

**Comment by CryZe** [comment]
Although I kind of have the opposite problem. Basically having native-client active on wasm makes it not compile.

**Comment by yoshuawuyts** [comment]
There's a private `with_client` method right now (doc hidden, publicly accessible tho) that will be opened up at some point. Wanted to make sure we indeed got the `http-client` interfaces right, but I think they're looking quite good which means we'll be able to expose them in the near future (and in turn make `surf` useful without native-client enabled.)
